### PR TITLE
feat: add API to publish community template

### PR DESCRIPTION
**Pull request by abm17**
## Description
Add API to publish community template

#### PR fixes following issue(s)
#26242 

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [X] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java [hunk 1]
```diff
@@ -196,6 +196,10 @@ public String getLastDeployedAt() {
     @JsonView(Views.Public.class)
     Boolean forkWithConfiguration;
 
+    // isCommunityTemplate represents whether this application has been published as a community template
+    @JsonView(Views.Public.class)
+    Boolean isCommunityTemplate;
```
Comment (nayan-rafiq) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java:201]: Why we need this field? Didn't find any usage of it.

Comment (abm17) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java:201]: This is used by the frontend to determine an application is published as a template

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**